sdk/sdk/src/main/java/org/apache/carbondata/sdk/file/CarbonIUD.java

```diff
@@ -121,6 +125,25 @@ private void closeDelete() throws IOException, InterruptedException {
     for (Map.Entry<String, Map<String, Set<String>>> path : this.filterColumnToValueMappingForDelete
         .entrySet()) {
       deleteExecution(path.getKey());
+      createEmptyMetadataFile(path.getKey());
+    }
+  }
+
+  private void createEmptyMetadataFile(String path) throws IOException {
+    if (!StringUtils.isEmpty(path)) {
+      // Get store path of the table.
+      String schemaFilePath = CarbonTablePath.getSchemaFilePath(path, configuration);
+      // If metadata folder exists, it is a transactional table
+      CarbonFile schemaFile = FileFactory.getCarbonFile(schemaFilePath, configuration);
+      boolean isTransactionalTable = schemaFile.exists();
+      if (!isTransactionalTable) {
```

Comment:
as it is in sdk module, it is always non-transactional right? I think we don't need this check here. Directly create an empty folder in the file path.

Reply:
yeah, you are right, earlier I was thinking about the operation performed on the transactional and not transactional table both.